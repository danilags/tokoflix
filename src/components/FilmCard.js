import React from 'react';
import { IMAGE_BASE_URL } from '../constants';

import {
  Card,
  CardBody,
  CardTitle,
  CardText,
} from 'reactstrap';

const FilmCard = (props) => (
  <Card style={styles.container}>
    <CardBody>
      <img style={{ width: '100%'  }} src={`${IMAGE_BASE_URL}${props.film.poster_path}`} alt="Card image cap" />
    </CardBody>
    <CardBody>
      <CardTitle>{props.film.title}</CardTitle>
      <CardText>
        <span style={{ padding: '2px' }}>{props.film.release_date} </span> | 
        <span style={{ padding: '2px' }}>{props.film.vote_average} </span> |
        <spn style={{ padding: '2px' }}>{props.film.adult ? 'Dewasa' : 'Remaja'}</spn>
      </CardText>
    </CardBody>
  </Card>
)
const styles = {
  container: {
    margin: '2rem 0',
  }
}

export default FilmCard;
