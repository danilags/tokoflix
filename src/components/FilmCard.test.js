import React from 'react';
import { mount } from 'enzyme';

import FilmCard from './FilmCard';

describe('<FilmCard />', () => {
  function onToggleFunc({}) {return {}};

  it('Component <FilmCard /> and matches snapShot', () => {
    let FilmCardProps = {
      film: {
        poster_path: 'http://anotherurl.com/7eTWQb475pM9H3NEOiUzGYASEx0.jpg' | null,
        title: 'Jurasic world',
        release_date: '2018-07-15',
        vote_average: 6.5,
        adult: true | false
      }
    };
    const wrapper = shallow(<FilmCard {...FilmCardProps} />)
    expect(wrapper).toMatchSnapshot();
  });

});