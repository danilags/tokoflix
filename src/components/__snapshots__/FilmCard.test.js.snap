// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmCard /> Component <FilmCard /> and matches snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmCard
    film={
      Object {
        "adult": 1,
        "poster_path": 0,
        "release_date": "2018-07-15",
        "title": "Jurasic world",
        "vote_average": 6.5,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardBody
          tag="div"
        >
          <img
            alt="Card image cap"
            src="https://image.tmdb.org/t/p/original/0"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </CardBody>,
        <CardBody
          tag="div"
        >
          <CardTitle
            tag="h5"
          >
            Jurasic world
          </CardTitle>
          <CardText
            tag="p"
          >
            <span
              style={
                Object {
                  "padding": "2px",
                }
              }
            >
              2018-07-15
               
            </span>
             |
            <span
              style={
                Object {
                  "padding": "2px",
                }
              }
            >
              6.5
               
            </span>
             |
            <spn
              style={
                Object {
                  "padding": "2px",
                }
              }
            >
              Dewasa
            </spn>
          </CardText>
        </CardBody>,
      ],
      "style": Object {
        "margin": "2rem 0",
      },
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <img
            alt="Card image cap"
            src="https://image.tmdb.org/t/p/original/0"
            style={
              Object {
                "width": "100%",
              }
            }
          />,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Card image cap",
            "src": "https://image.tmdb.org/t/p/original/0",
            "style": Object {
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardTitle
              tag="h5"
            >
              Jurasic world
            </CardTitle>,
            <CardText
              tag="p"
            >
              <span
                style={
                  Object {
                    "padding": "2px",
                  }
                }
              >
                2018-07-15
                 
              </span>
               |
              <span
                style={
                  Object {
                    "padding": "2px",
                  }
                }
              >
                6.5
                 
              </span>
               |
              <spn
                style={
                  Object {
                    "padding": "2px",
                  }
                }
              >
                Dewasa
              </spn>
            </CardText>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Jurasic world",
              "tag": "h5",
            },
            "ref": null,
            "rendered": "Jurasic world",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span
                  style={
                    Object {
                      "padding": "2px",
                    }
                  }
                >
                  2018-07-15
                   
                </span>,
                " |",
                <span
                  style={
                    Object {
                      "padding": "2px",
                    }
                  }
                >
                  6.5
                   
                </span>,
                " |",
                <spn
                  style={
                    Object {
                      "padding": "2px",
                    }
                  }
                >
                  Dewasa
                </spn>,
              ],
              "tag": "p",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2018-07-15",
                    " ",
                  ],
                  "style": Object {
                    "padding": "2px",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "2018-07-15",
                  " ",
                ],
                "type": "span",
              },
              " |",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    6.5,
                    " ",
                  ],
                  "style": Object {
                    "padding": "2px",
                  },
                },
                "ref": null,
                "rendered": Array [
                  6.5,
                  " ",
                ],
                "type": "span",
              },
              " |",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Dewasa",
                  "style": Object {
                    "padding": "2px",
                  },
                },
                "ref": null,
                "rendered": "Dewasa",
                "type": "spn",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardBody
            tag="div"
          >
            <img
              alt="Card image cap"
              src="https://image.tmdb.org/t/p/original/0"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </CardBody>,
          <CardBody
            tag="div"
          >
            <CardTitle
              tag="h5"
            >
              Jurasic world
            </CardTitle>
            <CardText
              tag="p"
            >
              <span
                style={
                  Object {
                    "padding": "2px",
                  }
                }
              >
                2018-07-15
                 
              </span>
               |
              <span
                style={
                  Object {
                    "padding": "2px",
                  }
                }
              >
                6.5
                 
              </span>
               |
              <spn
                style={
                  Object {
                    "padding": "2px",
                  }
                }
              >
                Dewasa
              </spn>
            </CardText>
          </CardBody>,
        ],
        "style": Object {
          "margin": "2rem 0",
        },
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <img
              alt="Card image cap"
              src="https://image.tmdb.org/t/p/original/0"
              style={
                Object {
                  "width": "100%",
                }
              }
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Card image cap",
              "src": "https://image.tmdb.org/t/p/original/0",
              "style": Object {
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardTitle
                tag="h5"
              >
                Jurasic world
              </CardTitle>,
              <CardText
                tag="p"
              >
                <span
                  style={
                    Object {
                      "padding": "2px",
                    }
                  }
                >
                  2018-07-15
                   
                </span>
                 |
                <span
                  style={
                    Object {
                      "padding": "2px",
                    }
                  }
                >
                  6.5
                   
                </span>
                 |
                <spn
                  style={
                    Object {
                      "padding": "2px",
                    }
                  }
                >
                  Dewasa
                </spn>
              </CardText>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Jurasic world",
                "tag": "h5",
              },
              "ref": null,
              "rendered": "Jurasic world",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <span
                    style={
                      Object {
                        "padding": "2px",
                      }
                    }
                  >
                    2018-07-15
                     
                  </span>,
                  " |",
                  <span
                    style={
                      Object {
                        "padding": "2px",
                      }
                    }
                  >
                    6.5
                     
                  </span>,
                  " |",
                  <spn
                    style={
                      Object {
                        "padding": "2px",
                      }
                    }
                  >
                    Dewasa
                  </spn>,
                ],
                "tag": "p",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "2018-07-15",
                      " ",
                    ],
                    "style": Object {
                      "padding": "2px",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    "2018-07-15",
                    " ",
                  ],
                  "type": "span",
                },
                " |",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      6.5,
                      " ",
                    ],
                    "style": Object {
                      "padding": "2px",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    6.5,
                    " ",
                  ],
                  "type": "span",
                },
                " |",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Dewasa",
                    "style": Object {
                      "padding": "2px",
                    },
                  },
                  "ref": null,
                  "rendered": "Dewasa",
                  "type": "spn",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
